intake_stac/catalog.py

```diff
@@ -230,6 +283,20 @@ class StacItem(AbstractStacCatalog):
     name = 'stac_item'
     _stac_cls = pystac.Item
 
+    def __getitem__(self, key):
+        result = super().__getitem__(key)
+        # TODO: handle non-string assets?
+        asset = self._entries[key]
+        storage_options = asset._stac_obj.extra_fields.get('xarray:storage_options', {})
+        open_kwargs = asset._stac_obj.extra_fields.get('xarray:open_kwargs', {})
+
+        if isinstance(result, DataSource):
+            kwargs = result._captured_init_kwargs
+            kwargs = {**kwargs, **dict(storage_options=storage_options), **open_kwargs}
+            result = result(*result._captured_init_args, **kwargs)
```

Comment:
@martindurant currently, `StacItem.__getitem__` will return a (subclass of) `DataSource`. Does this seem like the right way to control the parameters passed to that DataSource? If so, are `_captured_init_args` and `captured_init_kwargs` considered "public"?

Reply:
Just noting for posterity, intake-xarray's datasources define a `.kwargs` and `.storage_options` properties. We can't use those because they apparently aren't implemented by RasterIOSource.

Reply:
unfortunately i don't really follow this... i've always been a little confused about what should be handled by intake-xarray or whether intake-stac should just be stand-alone and define all the datasources under this repo. I sort of started down that road in https://github.com/intake/intake-stac/pull/75/files#diff-b45fa0c9c70f45ce9661f18946a5a2aed632ac4c1d3b1c09333291f77bbdfda6 but abandoned it...